src/libcore/iter/mod.rs

```diff
@@ -1643,20 +1708,145 @@ impl<I: DoubleEndedIterator, U, F> DoubleEndedIterator for FlatMap<I, U, F> wher
 ///
 /// [`fuse()`]: trait.Iterator.html#method.fuse
 /// [`Iterator`]: trait.Iterator.html
-#[derive(Clone, Debug)]
 #[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Fuse<I> {
-    iter: I,
-    done: bool
+    iter: <I as SpecFuse>::T,
```

Comment:
I suspect that this is going to change `Fuse<I>` from covariant in `I` to invariant in `I`: https://is.gd/btpbYx